android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonInstallationDialogFragment.kt

```diff
@@ -153,8 +154,12 @@ class AddonInstallationDialogFragment : AddonDialogFragment() {
         loadIcon(addon = addon, iconView = binding.icon)
 
         val allowedInPrivateBrowsing = rootView.findViewById<AppCompatCheckBox>(R.id.allow_in_private_browsing)
-        allowedInPrivateBrowsing.setOnCheckedChangeListener { _, isChecked ->
-            allowPrivateBrowsing = isChecked
+        if (addon.isIncognitoNotAllowed()) {
+            allowedInPrivateBrowsing.isVisible = false
+        } else {
+            allowedInPrivateBrowsing.setOnCheckedChangeListener { _, isChecked ->
+                allowPrivateBrowsing = isChecked
+            }
```

Comment:
Having an `enum` (see [this comment](https://github.com/mozilla-mobile/firefox-android/pull/5571/files#r1488182414)) will make the interaction much easier, so we won't need `isIncognitoNotAllowed`.